src/VisualStudio/VisualStudioDiagnosticsToolWindow/OptionPages/InternalOptionsControl.cs

```diff
@@ -19,28 +19,83 @@ public InternalOptionsControl(string featureOptionName, IServiceProvider service
         {
             _featureOptionName = featureOptionName;
 
-            var panel = new StackPanel();
-            this.AddOptions(panel);
+            // options
+            var optionsPanel = new StackPanel();
+            this.AddOptions(optionsPanel);
 
-            var viewer = new ScrollViewer();
-            viewer.VerticalScrollBarVisibility = ScrollBarVisibility.Auto;
-            viewer.HorizontalScrollBarVisibility = ScrollBarVisibility.Auto;
+            // scroll
+            var viewer = new ScrollViewer()
+            {
+                VerticalScrollBarVisibility = ScrollBarVisibility.Auto,
+                HorizontalScrollBarVisibility = ScrollBarVisibility.Auto
+            };
+
+            viewer.Content = optionsPanel;
+
+            // search 
+            var searchBox = new TextBox() { MinWidth = 200, HorizontalAlignment = HorizontalAlignment.Stretch };
+
+            var searchButton = new Button() { Content = "Search" };
+            searchButton.Click += (o, a) =>
+            {
+                foreach (var item in optionsPanel.Children.OfType<CheckBox>())
+                {
+                    var title = item.Content as string;
+                    if (title == null)
+                    {
+                        continue;
+                    }
+
+                    // pattern not match
+                    if (title.IndexOf(searchBox.Text, StringComparison.OrdinalIgnoreCase) < 0)
+                    {
+                        // hide it
+                        item.Visibility = Visibility.Collapsed;
```

Comment:
Is this implicitly ANDing with the previous search?

Reply:
@jasonmalinowski I wasn't care much whether it is Anding with previous result or not since there is "clear" button. but sure, I can make it not to "And" if previous search.

Reply:
I shouldn't have to clear to make it work like every other search box known to humanity. :smile: